tests/integration/tests/test_version_upgrades.py

```diff
@@ -0,0 +1,59 @@
+#
+# Copyright 2024 Canonical, Ltd.
+#
+import logging
+from typing import List
+
+import pytest
+from test_util import config, harness, util
+
+LOG = logging.getLogger(__name__)
+
+
+@pytest.mark.node_count(3)
+@pytest.mark.no_setup()
+@pytest.mark.skipif(
+    config.VERSION_UPGRADE_CHANNELS is None, reason="No upgrade channels configured"
+)
+def test_version_upgrades(instances: List[harness.Instance]):
+    channels = config.VERSION_UPGRADE_CHANNELS.splitlines()
```

Comment:
if this const was a list, you wouldn't have to split this here...

```suggestion
    channels = config.VERSION_UPGRADE_CHANNELS
```